src/core_plugins/kibana/public/context/app.js

```diff
@@ -0,0 +1,100 @@
+import _ from 'lodash';
+
+import uiModules from 'ui/modules';
+import contextAppTemplate from './app.html';
+import './components/loading_button';
+import './components/size_picker/size_picker';
+import {
+  createInitialQueryParametersState,
+  QueryParameterActionsProvider,
+  QUERY_PARAMETER_KEYS,
+} from './query_parameters';
+import { QueryActionsProvider } from './query';
+
+const module = uiModules.get('apps/context', [
+  'elasticsearch',
+  'kibana',
+  'kibana/config',
+  'kibana/notify',
+  'ngRoute',
+]);
+
+module.directive('contextApp', function ContextApp() {
+  return {
+    bindToController: true,
+    controller: ContextAppController,
+    controllerAs: 'contextApp',
+    restrict: 'E',
+    scope: {
+      anchorUid: '=',
+      columns: '=',
+      indexPattern: '=',
+      predecessorCount: '=',
+      successorCount: '=',
+      sort: '=',
+      discoverUrl: '=',
+    },
+    template: contextAppTemplate,
+  };
+});
+
+function ContextAppController($scope, Private) {
+  const queryParameterActions = Private(QueryParameterActionsProvider);
+  const queryActions = Private(QueryActionsProvider);
+
+  this.state = createInitialState(this.discoverUrl);
+
+  this.actions = _.mapValues(Object.assign(
+    {},
+    queryParameterActions,
+    queryActions,
+  ), (action) => (...args) => action(this.state)(...args));
```

Comment:
Well, here's my thought process.

1. To me, our chief priority should be that someone reading this code needs to be able to understand it. So, I posit that displaying each action as an explicit reference is transparent, vs. a map over references is opaque, and it sounds like we agree on this point.

2. If it turns out there are so many actions that even _this_ becomes difficult to read, then I think we have a couple options. The less invasive option would be to group actions into types of actions, e.g.:

```javascript
this.rowActions = {
    setAllRows: queryActions.setAllRows(this.state),
    fetchAllRowsWithNewQueryParameters: queryActions.fetchAllRowsWithNewQueryParameters(this.state),
};

this.navigationActions = {
    fetchMorePredecessorRows: queryActions.fetchMorePredecessorRows(this.state),
    fetchGivenPredecessorRows: queryActions.fetchGivenPredecessorRows(this.state),
    fetchMoreSuccessorRows: queryActions.fetchMoreSuccessorRows(this.state),
    fetchGivenSuccessorRows: queryActions.fetchGivenSuccessorRows(this.state),
};
```

This is how I like to tackle scalability with folder structures too. I group cohesive files into a single folder, treating them as a single module. As this module's responsibilities grow, the number of files within the folder grow. Once this folder grows to the point where it becomes difficult to understand, I reorganize the files within it into submodules, each of which has their own subfolder.

It's also possible that if there are a ton of actions in this file then there is a deeper architectural issue. For example, maybe this file would need to to be split up into separate directives, which have access to their own actions.

3. What do you mean by DRY in this case? Where is the repetition? In other files?